src/internal/operators/bufferTime.ts

```diff
@@ -144,8 +146,14 @@ class BufferTimeSubscriber<T> extends Subscriber<T> {
       const timeSpanOnlyState = { subscriber: this, context, bufferTimeSpan };
       this.add(context.closeAction = scheduler.schedule(dispatchBufferTimeSpanOnly, bufferTimeSpan, timeSpanOnlyState));
     } else {
-      const closeState = { subscriber: this, context };
-      const creationState: DispatchCreateArg<T> = { bufferTimeSpan, bufferCreationInterval, subscriber: this, scheduler };
+      const closeState = { subscriber: this, context, destination: this.destination };
+      const creationState: DispatchCreateArg<T> = {
```

Comment:
Why aren't we using idiomatic JS like closure state? Passing around the explicit state does not seem ideal.